biojava-genome/src/main/java/org/biojava/nbio/genome/util/ChromosomeMappingTools.java

```diff
@@ -0,0 +1,1394 @@
+
+            if (start +1 <= cdsStart +1 && end >= cdsStart+1) {
+
+                if (end >= chromPos) {
+                    // we are reaching our target position
+                    // -1 is important  here...
+                    int tmp = chromPos - cdsStart -1;
+
+                    if ( debug) {
+                        System.out.println("cdl:" + codingLength + " | " + tmp);
+                        System.out.println(" -> found position in UTR exon:  " + chromPos + " " + format(start) + " " + format(tmp) + " " + cdsStart + " " + codingLength);
+                    }
+                    return codingLength + tmp;
+                }
+                inCoding = true;
+                codingLength += (end - cdsStart);
+                if (debug) {
+                    System.out.println("     UTR         : " + format(start) + " - " + (cdsStart ));
+                    System.out.println(" ->  Exon        : " + format(cdsStart+1) + " - " + format(end) + " | " + format(end - cdsStart) + " | " + codingLength + " | " + (codingLength % 3));
+                }
+            } else if (start <= cdsEnd && end >= cdsEnd) {
+                //System.out.println(" <-- CDS end at: " + cdsEnd );
+                inCoding = false;
+                if (cdsEnd >= chromPos && (start +1 <= chromPos)) {
+                    int tmp =  chromPos - start -1 ;
+
+                    if ( debug)
+                        System.out.println(" -> cdsForward found position in non coding exon#"+i+":  " + chromPos + " " + format(start+1) + " " + format(tmp) + " " + cdsStart   + " " + codingLength);
+                    return codingLength + tmp ;
+                }
+                codingLength += (cdsEnd - start);
+                if (debug) {
+                    System.out.println(" <-  Exon        : " + format(start+1) + " - " + format(cdsEnd) + " | " + format(cdsEnd - start+1) + " | " + codingLength + " | " + (codingLength % 3));
+                    System.out.println("     UTR         : " + format(cdsEnd + 1) + " - " + format(end));
+                }
+
+            } else if (inCoding) {
+
+                if (end >= chromPos && (start +1 <=chromPos)) {
+
+                    int tmp = chromPos-start-1 ;
+
+                    if ( debug) {
+                        System.out.println(codingLength + " | " + tmp);
+                        System.out.println(" -> found position in coding exon #" + (i + 1) + ":  " + format(chromPos) + " " + format(start + 1) + " " + format(tmp) + " " + cdsStart + " " + codingLength);
+                    }
+
+                    return codingLength + tmp ;
+                }
+                // full exon is coding
+                codingLength += (end - start);
+                if (debug)
+                    System.out.println("     Exon        :" + format(start) + " - " + format(end) + " | " + format(end - start) + " | " + codingLength + " | " + (codingLength % 3));
```

Comment:
All these should be done with `logger.debug` and not use a `debug` constant at all.